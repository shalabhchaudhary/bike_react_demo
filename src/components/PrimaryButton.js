import React from "react";
import { Text, TouchableOpacity } from "react-native";


const PrimaryButton = (props)=>{
    return(
        <TouchableOpacity style={{
            padding: 10,
            backgroundColor:props.bgColor,
            marginStart: 44,
            marginTop:24,
            marginEnd:20, 
            width: 350,
            height:54,
            alignItems:'center',
            justifyContent:'center',
            borderRadius: 12}} 
            activeOpacity = {1}
            onPress={props.onClick}>
            <Text
                style={{color:'#ffffff', textAlign: 'center', fontWeight: 'bold'}}>
                {props.label}   
            </Text>
        </TouchableOpacity>
    );
}

export default PrimaryButton;