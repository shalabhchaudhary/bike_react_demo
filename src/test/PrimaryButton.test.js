import React from "react";
import PrimaryButton from "./PrimaryButton";
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

test('Checking Primary Button', () => {
    const tree = renderer.create(<PrimaryButton styles={{color:'#7F23D3',alignItems:'center',
    justifyContent:'center'}}>Login</PrimaryButton>);
    expect(tree).toMatchSnapshot();
});

// test('onPress', () => {
//     let i = 0
//     const onPress = () => i++
//     const wrapperPress = shallow(<PrimaryButton onPress={onPress} text='hi' />)
  
//     expect(wrapperPress.prop('onPress')).toBe(onPress) // uses the right handler
//     expect(i).toBe(0)
//     wrapperPress.simulate('press')
//     expect(i).toBe(1)
//   });