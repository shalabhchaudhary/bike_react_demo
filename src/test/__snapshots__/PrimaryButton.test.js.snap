// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking Primary Button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderRadius": 12,
      "height": 54,
      "justifyContent": "center",
      "marginEnd": 20,
      "marginStart": 44,
      "marginTop": 24,
      "opacity": 1,
      "padding": 10,
      "width": 350,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  />
</View>
`;
